### ISSUE TITLE
[FR]disable threads for wasm?

### ISSUE THREAD

**Issue by trcrsired** [google/benchmark#1774]
i try to use google benchmark for wasm and it does not provide thread. Is that possible to use this without pthread?

**Comment by dmah42** [comment]
https://github.com/google/benchmark/blob/main/CMakeLists.txt#L324  we expect to have pthreads available

**Comment by LebedevRI** [comment]
I guess we only really need threading for `->Threads()` feature, so we could make it disable-ableable. Do we want that?

**Comment by justzh** [comment]
No, you don't disable threads for `wasm`; disable threads for meta-data.

**Comment by justzh** [comment]
Thoughts?

**Comment by LebedevRI** [comment]
I'm afraid i'm failing to parse last two comments.
The topic-starter is asking about being able to build the library without threading,
and i've said that we could theoretically do that since we don't strictly need them always.

**Comment by justzh** [comment]
hai

**Comment by justzh** [comment]
I think we want that, um, uhhhhhh, yeah..

**Comment by justzh** [comment]
We don't need this customer.

**Comment by justzh** [comment]
Sorry, but it's true.
